net/interfaces/interfaces.go

```diff
@@ -182,3 +183,32 @@ var linkLocalIPv4 = func() *net.IPNet {
 	}
 	return ipNet
 }()
+
+// State is intended to store the state of the machine's network interfaces,
+// routing table, and other network configuration.
+// For now it's pretty basic.
+type State struct {
+	InterfaceIPs map[string][]net.IP
+}
+
+func (s *State) Equal(s2 *State) bool {
+	return reflect.DeepEqual(s, s2)
+}
+
+func (s *State) RemoveTailscaleInterfaces() {
+	for name := range s.InterfaceIPs {
+		if strings.HasPrefix(name, "tailscale") {
```

Comment:
A TODO here for using the `--tun` flag value?

OpenBSD has a different `DefaultTunName`. And in a ridiculous world, some other tailscaled's tun interface might be the packet default route.

Reply:
Done.